WordPress/openverse-frontend
# Add skip to content button

## Fixes


Fixes #659 by @sarayourfriend

## Description


Adds a "Skip to content" button on the search page that should be the _first_ thing tabbable on the page. This will make it the first thing that keyboard navigation users will tab into and therefore make it easy to skip to the search results.

This is a general purpose solution that can be applied to all pages that need it eventually (for example the information pages or single result pages). It tries to be pretty unassuming in how it does things while also making it easy to add to pages in the future. For now we can put it just on the search page and see how it works out. It's the most critical page that needs this anyway.

## Testing Instructions

Do the following both with and without a screen reader using keyboard navigation as well as using mouse navigation and verify that the interactions make sense:

1. Execute a search from the home page using the mouse and verify that the experience still works and makes sense
2. Execute a search from the home page using keyboard navigation and verify the when you hit tab on the search results page the "Skip to content" button is focused and that it works to focus you into the results
3. Execute a search from the home page using a screen reader and keyboard navigation and verify that the "Skip to content" button is either automatically focused by the screen reader or is the first element focused when using the screen reader controls to navigate the page

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
